File: src/main/java/com/fifo/ticketing/domain/book/dto/BookCompleteDto.java
package com.fifo.ticketing.domain.book.dto;

import com.fifo.ticketing.domain.seat.entity.SeatStatus;
import java.time.LocalDateTime;
import java.util.List;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Getter
@Builder
public class BookCompleteDto {

    private Long performanceId;
    private String performanceTitle;
    private String encodedFileName;
    private LocalDateTime performanceStartTime;
    private LocalDateTime performanceEndTime;
    private String placeName;
    private List<BookSeatViewDto> seats;
    private int totalPrice;
    private int quantity;
    @Setter
    private boolean paymentCompleted;
    private String urlPrefix;

    public String getUrl() {
        return urlPrefix + encodedFileName;
    }

}

File: src/main/java/com/fifo/ticketing/domain/book/dto/BookCompleteDto.kt
package com.fifo.ticketing.domain.book.dto

import lombok.Setter
import java.time.LocalDateTime

data class BookCompleteDto(
    val performanceId: Long,
    val performanceTitle: String,
    val encodedFileName: String,
    val performanceStartTime: LocalDateTime,
    val performanceEndTime: LocalDateTime,
    val placeName: String,
    val seats: List<BookSeatViewDto>,
    val totalPrice: Int,
    val quantity: Int,
    var paymentCompleted: Boolean = false,

    var urlPrefix: String
) {
    val url: String
        get() = "$urlPrefix$encodedFileName"
}